### PR TITLE
build(bazel): run aio firebase redirect tests under bazel

### PR DESCRIPTION
**Pull request by kormide**
@gkalpak @devversion @josephperrott @gregmagolan 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?

The firebase redirects test now runs under bazel/ci.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -350,8 +350,6 @@ jobs:
       - run: yarn --cwd aio test-a11y-score-localhost
         # Check the bundle sizes.
       - run: yarn --cwd aio payload-size
-        # Run unit tests for Firebase redirects
-      - run: yarn --cwd aio redirects-test
```
Comment (kormide) [inline .circleci/config.yml:354]: This is now run as part of the catch-all `test:ci` yarn script.

File: aio/tools/firebase-test-utils/FirebaseRedirectSource.spec.ts [hunk 2]
```diff
@@ -361,5 +361,5 @@ function testSourceMatch(
     matches: { [url: string]: unknown|undefined }) {
   expect(source.namedGroups).toEqual(captures.named || []);
   expect(source.restNamedGroups).toEqual(captures.rest || []);
-  Object.keys(matches).forEach(url => expect(source.match(url)).toEqual(matches[url]));
+  Object.keys(matches).forEach(url => expect(source.match(url) as any).toEqual(matches[url]));
```
Comment (kormide) [inline aio/tools/firebase-test-utils/FirebaseRedirectSource.spec.ts:364]: I was getting a type error here.

File: aio/tools/firebase-test-utils/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,16 @@
+load("@aio_npm//@bazel/jasmine:index.bzl", "jasmine_node_test")
+
+jasmine_node_test(
+    name = "test",
+    srcs = [
+        "test.js",
+        "tsconfig.json",
+        "//aio:tsconfig.json",
+    ] + glob(["*.ts"]),
```
Comment (devversion) [inline aio/tools/firebase-test-utils/BUILD.bazel:9]: Should the TS and json files be part of `data` instead? Maybe related to the TS `as any` cast error

**Comment by kormide** [comment]
Changed to a `nodejs_test` to use the runner script as originally intended. That also fixed the type error.

File: .circleci/config.yml (hunk 1 above)
Comment (kormide) [inline .circleci/config.yml:354]: This is now covered under `yarn test:ci`.

**Comment by kormide** [comment]
Updated to include the deployment unit tests.

File: packages/service-worker/config/BUILD.bazel [hunk 4]
```diff
@@ -7,11 +7,16 @@ exports_files([
...
     srcs = glob([
         "*.ts",
         "src/**/*.ts",
     ]),
+)
+
+ng_module(
+    name = "config",
+    srcs = [":sources"],
     deps = ["//packages/core"],
```
Comment (gkalpak) [inline packages/service-worker/config/BUILD.bazel:21]: Not directly related to this PR, but why is `//packages/core` needed?

**Review by gkalpak — APPROVED** [review]

File: packages/service-worker/config/BUILD.bazel (hunk 4 above)
Comment (kormide) [reply packages/service-worker/config/BUILD.bazel:21]: Removing it doesn't cause anything to fail, so probably accidental.

**Review by josephperrott — APPROVED** [review]
LGTM

**Pull request merged**